### ISSUE TITLE
How i can change size of grid card

### ISSUE THREAD

**Issue by amasahara** [liodali/checkbox_grouped#50]
I try create some grid card but i want grid is more size width but i don't know how.

**Comment by liodali** [comment]
can you provide me example of your widget and how many item you want per column or size that you want to apply and i will help to do that

**Comment by amasahara** [comment]
![image](https://user-images.githubusercontent.com/5392719/124289080-1cc48180-db7c-11eb-8e6b-1304363c18aa.png)
i want it more width because i don't want wrap text and don't need overflow

**Comment by liodali** [comment]
give me example of code that you are using for that page

**Comment by amasahara** [comment]
Widget createQuyenItem(String tenNhomPhanQuyen, List<PermissionModel> lst) {
    return Flexible(
      fit: FlexFit.tight,
      child: CustomGroupedCheckbox<PermissionModel>.grid(
        controller: CustomGroupController(isMultipleSelection: true),
        groupTitle: Container(
          padding: EdgeInsets.all(5.0),
          child: Text(tenNhomPhanQuyen),
        ),
        itemBuilder: (ctx, index, selected, isDisabled) {
          return Container(
            width: 600,
            height: 300,
            child: Card(
              margin: EdgeInsets.only(
                top: 5.0,
                bottom: 5.0,
              ),
              child: Stack(
                children: [
                  Center(
                    child: ListTile(
                      tileColor: Colors.white,
                      title: Text(
                        lst[index].showName,
                        maxLines: 3,
                        overflow: TextOverflow.ellipsis,
                      ),
                    ),
                  ),
                  Positioned(
                    left: 10,
                    right: 10,
                    bottom: 0,
                    child: AnimatedContainer(
                      duration: Duration(milliseconds: 300),
                      color: Colors.green[300],
                      height: selected ? 20 : 0,
                    ),
                  ),
                ],
              ),
            ),
          );
        },
        gridDelegate: const SliverGridDelegateWithFixedCrossAxisCount(
          crossAxisCount: 4,
        ),
        values: lst,
      ),
    );
  }

**Comment by liodali** [comment]
how many item you want per each column 
the size that you put it using container will not be considerated if you already put `crossAxisCount`  = 4

**Comment by amasahara** [comment]
yes i want 4 item of each

**Comment by amasahara** [comment]
hope you can make it show like this but with more column
![image](https://user-images.githubusercontent.com/5392719/124295702-5947ab80-db83-11eb-8fbc-29b8ab772274.png)

**Comment by liodali** [comment]
you want to make screen like that !

**Comment by amasahara** [comment]
yes

**Comment by liodali** [comment]
you want to create  list which each item is already groupCheckbox ?

**Comment by liodali** [comment]
I will create small example that  illustrate that image

**Comment by amasahara** [comment]
yes, it like i want manager some role of application, and show it to group checkbox of each role type

**Comment by liodali** [comment]
see this example 
https://github.com/liodali/checkbox_grouped/blob/master/example/lib/grid_of_grouped_checkbox.dart

**Comment by amasahara** [comment]
Could you make an example that use model. i meet problem with it. Null check operator used on a null value
My code:
`SingleChildScrollView(
                child: GridView.builder(
                    gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
                      crossAxisCount: 2,
                      mainAxisExtent: 368,
                      mainAxisSpacing: 5.0,
                    ),
                    itemCount: size,
                    shrinkWrap: true,
                    itemBuilder: (ctx, index) {
                      final lstString = <String>[];
                      return DefaultTextStyle(
                        style: TextStyle(fontSize: 11),
                        child: SimpleGroupedCheckbox(
                          controller: listOfController[index],
                          groupTitle: 'User manager',
                          helperGroupTitle: true,
                          values: List.generate(2, (i) {
                            if (i == 0) {
                              return List.generate(
                                  controller.dsQuyen
                                      .where((p) => p.name.contains('user'))
                                      .length,
                                  (j) => controller.dsQuyen
                                      .where((p) => p.name.contains('user'))
                                      .toList()[j]);
                            } else if (i == 1) {
                              return List.generate(
                                  controller.dsQuyen
                                      .where((p) => p.name.contains('role'))
                                      .length,
                                  (j) => controller.dsQuyen
                                      .where((p) => p.name.contains('role'))
                                      .toList()[j]);
                            }
                          }),
                          itemsTitle: List.generate(2, (i) {
                            if (i == 0) {
                              controller.dsQuyen
                                  .where((p) => p.name.contains('user-'))
                                  .toList()
                                  .forEach((element) {
                                lstString.add(element.showName);
                              });
                            } else if (i == 1) {
                              controller.dsQuyen
                                  .where((p) => p.name.contains('role-'))
                                  .toList()
                                  .forEach((element) {
                                lstString.add(element.showName);
                              });
                            }

                            return;
                          }),
                          isLeading: true,
                        ),
                      );
                    }),
              ),`

**Comment by liodali** [comment]
ok i will change the example it will be available tomorrow

**Comment by liodali** [comment]
check the example 
https://github.com/liodali/checkbox_grouped/blob/master/example/lib/grid_of_grouped_checkbox.dart
if the problem persisten it will nice to share more information  like console log or line that cause the problem

**Comment by amasahara** [comment]
maybe way of my database design different with faker. my data result is
PermissionModel(id,   name,        showName)
PermissionModel(1, user-view, View User List), 
PermissionModel(2, user-edit, Edit User), 
PermissionModel(3, user-delete, Delete user), 
PermissionModel(4, user-add, Add user)

PermissionModel(5, role-view, View Role List), 
PermissionModel(6, role-edit, Edit Role ), 
PermissionModel(7, role-delete, Delete Role ), 
PermissionModel(8, role-add, Add Role )
it have no grouptile name, i base on name to set group title for each group ex: user- is User Manager or role- is Role Manager and then show it in group

**Comment by liodali** [comment]
you work on list directly ! or you group them !

**Comment by amasahara** [comment]
i use list directly

**Comment by liodali** [comment]
why not you group them by name and you will have group of list by name
compute your data
```dart
data.groupBy((e)=>(e.name as String).split("-")[0])
```
groupBy method
```dart
extension Iterables<E> on Iterable<E> {
  Map<K, List<E>> groupBy<K>(K Function(E) keyFunction) => fold(
      <K, List<E>>{},
      (Map<K, List<E>> map, E element) =>
          map..putIfAbsent(keyFunction(element), () => <E>[]).add(element));
}

```

**Comment by amasahara** [comment]
Thanks you very much for your help. I has been do it already. this is my code hope can help some one if needed and i have 1 question that i don't understand height of container why you multiplication with 5.1

`Container(
                  height: 250 * 5.1,
                  child: Column(
                    mainAxisSize: MainAxisSize.max,
                    children: [
                      Expanded(
                        child: GridView.builder(
                            physics: NeverScrollableScrollPhysics(),
                            gridDelegate:
                                SliverGridDelegateWithFixedCrossAxisCount(
                              crossAxisCount: 4,
                              mainAxisExtent: 250,
                              mainAxisSpacing: 1.0,
                            ),
                            itemCount: size,
                            shrinkWrap: true,
                            itemBuilder: (ctx, index) {
                              String groupTitle = "";
                              List<PermissionModel> lstPermissionModel =
                                  <PermissionModel>[];
                              if (index == 0) {
                                lstPermissionModel = controller.dsQuyen
                                    .where((p) => p.name.contains('user'))
                                    .toList();
                                groupTitle = "Quản lý người dùng";
                              } else if (index == 1) {
                                lstPermissionModel = controller.dsQuyen
                                    .where((p) => p.name.contains('role'))
                                    .toList();
                                groupTitle = "Quản lý phân quyền";
                              } else if (index == 2) {
                                lstPermissionModel = controller.dsQuyen
                                    .where((p) => p.name.contains('hoa-don'))
                                    .toList();
                                groupTitle = "Quản lý hóa đơn";
                              } else if (index == 3) {
                                lstPermissionModel = controller.dsQuyen
                                    .where((p) => p.name.contains('ban'))
                                    .toList();
                                groupTitle = "Quản lý bàn";
                              } else if (index == 4) {
                                lstPermissionModel = controller.dsQuyen
                                    .where((p) => p.name.contains('tang'))
                                    .toList();
                                groupTitle = "Quản lý tầng";
                              }

                              return DefaultTextStyle(
                                style: TextStyle(fontSize: 11),
                                child: SimpleGroupedCheckbox<PermissionModel>(
                                  controller: listOfController[index],
                                  groupTitle: groupTitle,
                                  helperGroupTitle: true,
                                  values: lstPermissionModel,
                                  itemsTitle: lstPermissionModel
                                      .map((e) => e.showName)
                                      .toList(),
                                  isLeading: true,
                                ),
                              );
                            }),
                      ),
                    ],
                  ),
                ),`

**Comment by liodali** [comment]
i will give gist that contain small help to overcome your problem 
and for 5.1 , .1 is just to get more space for button and 5 is the  size of list divided by 2 ( number of item per column )

**Comment by amasahara** [comment]
i meet another problem that i want load an list for checked value in this list like update roles. i try but don't know which way to do that

**Comment by liodali** [comment]
you want to set checked values programmatically

**Comment by liodali** [comment]
look at this gist
https://gist.github.com/liodali/f2fdbbd272bfd49bd95e71865cdeefff

**Comment by amasahara** [comment]
yes i want set checked values programmatically. Exam: i have another list of permission that has been checked and i want set it checked when show form

**Comment by liodali** [comment]
you can use GroupController and set initvalues for now 
until i will make that feature available
